tensorflow/compiler/mlir/tensorflow/ir/tf_ops_a_m.cc

```diff
@@ -1129,80 +1131,50 @@ LogicalResult ConstOp::inferReturnTypes(
 // Conv2DOp and Conv3DOp
 //===----------------------------------------------------------------------===//
 
-template <typename OpT>
-static LogicalResult VerifyConvOpAttributes(OpT op, int num_dims) {
-  if (!IsOfRankOrUnranked(op.getResult(), num_dims))
-    return op.emitOpError()
-           << "requires result to be " << num_dims << "D tensor";
-
+static LogicalResult VerifyConvOpAttributesV2(
+    int num_dims, ArrayRef<Attribute> strides, ArrayRef<Attribute> dilations,
+    llvm::Optional<mlir::Location> location) {
+  int64_t strides_size = strides.size();
+  if (strides_size != num_dims)
+    return emitOptionalError(
+        location, "requires strides attribute length to be ", num_dims);
   auto is_not_positive = [](Attribute val) {
     return val.cast<IntegerAttr>().getValue().getSExtValue() <= 0;
   };
+  if (llvm::any_of(strides, is_not_positive))
+    return emitOptionalError(location, "requires positive strides");
 
-  int64_t strides_size = op.strides().size();
-  if (strides_size != num_dims)
-    return op.emitOpError() << "requires strides attribute length to be "
-                            << num_dims << "; actual length " << strides_size;
-  if (llvm::any_of(op.strides().getValue(), is_not_positive))
-    return op.emitOpError("requires positive strides");
-
-  int64_t dilations_size = op.strides().size();
-  if (op.dilations().size() != num_dims)
-    return op.emitOpError() << "requires dilations attribute length to be "
-                            << num_dims << "; actual length " << dilations_size;
-  if (llvm::any_of(op.dilations().getValue(), is_not_positive))
-    return op.emitOpError("requires positive dilations");
+  int64_t dilations_size = dilations.size();
+  if (dilations_size != num_dims)
+    return emitOptionalError(
+        location, "requires dilations attribute length to be ", num_dims);
+  if (llvm::any_of(dilations, is_not_positive))
+    return emitOptionalError(location, "requires positive dilations");
 
   return success();
 }
 
+template <typename OpT>
+static LogicalResult VerifyConvOpAttributes(OpT op, int num_dims) {
+  if (!IsOfRankOrUnranked(op.getResult(), num_dims))
```

Comment:
We shouldn't need to verify the result type explicitly as it is done within your method. So, we can rename VerifyConvOpAttributesV2 as VerifyConvOpAttributes and only have one function to verify attributes.

Reply:
Hi @smit-hinsu I have made two versions as ``conv2DBackpropInputOp`` is also using ``verifyConvOpAttributes``, and there the result type is not checked.

Reply:
This seems it hasn't been changed and we still have VerifyConvOpAttributesV2 andd VerifyConvOpAttributes.

Reply:
As you mentioned earlier, we will now need to add result type rank check for conv2DBackpropInputOp. Right?